contracts/base/strategies/multi/AaveMaiBalStrategyBase.sol

```diff
@@ -0,0 +1,142 @@
+* extent permissible pursuant to applicable law any and all liability as well
+* as all warranties, including any fitness for a particular purpose with respect
+* to Tetu and/or the underlying software and the use thereof are disclaimed.
+*/
+
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+import "./../StrategyBase.sol";
+
+import "../../../third_party/uniswap/IWETH.sol";
+import "./pipelines/LinearPipeline.sol";
+import "./pipes/UnwrappingPipe.sol";
+import "./pipes/AaveWethPipe.sol";
+import "./pipes/MaiCamWMaticPipe.sol";
+import "./pipes/MaiStablecoinPipe.sol";
+import "./pipes/BalVaultPipe.sol";
+
+/// @title AAVE->MAI->BAL Multi Strategy
+/// @author belbix, bogdoslav
+contract AaveMaiBalStrategyBase is StrategyBase, LinearPipeline {
+    /// @notice Strategy type for statistical purposes
+    string public constant override STRATEGY_NAME = "AaveMaiBalStrategyBase";
+    /// @notice Version of the contract
+    /// @dev Should be incremented when contract changed
+    string public constant VERSION = "1.0.0";
+    /// @dev Placeholder, for non full buyback need to implement liquidation
+    uint256 private constant _BUY_BACK_RATIO = 10000;
+
+    /// @dev Assets should reflect underlying tokens for investing
+    address[] private _assets;
+
+    uint256 private _totalAmount = 0;
+
+    address public WMATIC;
+
+    /// @notice Contract constructor
+    constructor(
+        address _controller,
+        address _underlying,
+        address _vault,
+        address[] memory __rewardTokens,
+        address _WMATIC,
+        AaveWethPipeData memory aaveWethPipeData,
+        MaiCamWMaticPipeData memory maiCamWMaticPipeData,
+        MaiStablecoinPipeData memory maiStablecoinPipeData,
+        BalVaultPipeData memory balVaultPipeData
+    ) StrategyBase(_controller, _underlying, _vault, __rewardTokens, _BUY_BACK_RATIO)
+    {
+        WMATIC = _WMATIC;
+        require(_underlying == WMATIC, "MS: underlying must be WMATIC");
+        _rewardTokens = __rewardTokens;
+
+        // Build pipeline
+        addPipe(new UnwrappingPipe(WMATIC));
+        addPipe(new AaveWethPipe(aaveWethPipeData));
+        addPipe(new MaiCamWMaticPipe(maiCamWMaticPipeData));
+        addPipe(new MaiStablecoinPipe(maiStablecoinPipeData));
+        addPipe(new BalVaultPipe(balVaultPipeData));
+
+        console.log('Initialized+++');
+    }
+
+  /*  /// @dev creates segment and initializes its context //TODO move this function to appropriate base file
+    function initSegment(Pipe pipe, bytes memory context) internal returns (PipeSegment memory segment) {
+        segment = PipeSegment(pipe, context);
+        segment.context = segment.init(); //TODO
+    }*/
+
+
+/// @dev Stub function for Strategy Base implementation
+    function rewardPoolBalance() public override pure returns (uint256 bal) {
+        bal = 0;
```

Comment:
no way, need to implement it very accurate
it uses in vault logic as general criticalattribute